### PR TITLE
Jc 8

### PR DESCRIPTION
**Pull request by aalexeen**
attempt to merge JC-8 to dev

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        <!-- https://mvnrepository.com/artifact/org.postgresql/postgresql -->
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>${postgresql.version}</version>
+        </dependency>
+
+        <!-- https://mvnrepository.com/artifact/info.picocli/picocli -->
+        <dependency>
+            <groupId>info.picocli</groupId>
```
Comment (pelmegov) [inline pom.xml]: Is this required to implement the feature?

File: src/main/java/org/jacp/competitionservice/CompetitionserviceApplication.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.jacp.competitionservice;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class CompetitionserviceApplication {
+
+	public static void main(String[] args) {
```
Comment (pelmegov) [inline src/main/java/org/jacp/competitionservice/CompetitionserviceApplication.java]: Please setup your editor to use 4 spaces instead tabs.

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,10 @@
+spring.datasource.url=jdbc:postgresql://${DB_HOST:localhost}:5432/competitionservice
```
Comment (pelmegov) [inline src/main/resources/application.properties]: Yaml format is more like a standard in the industry, so let's use it.

File: src/main/resources/db/changelog/init/01-create-competition.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-table-competition
+      author: Aleksandr
+      changes:
+        - createTable:
+            tableName: competition
+            columns:
```
Comment (pelmegov) [inline src/main/resources/db/changelog/init/01-create-competition.yaml:8]: Competition can have:
1. start date
2. end date
3. participants
4. status
5. tasks
And potentially more. 

Could you please explore a comprehensive list of fields that could potentially be configured?

File: src/main/resources/db/competition.csv [hunk 5]
```diff
@@ -0,0 +1,34 @@
+Name;Description
+Competition1;test description of competition1
```
Comment (pelmegov) [inline src/main/resources/db/competition.csv]: It's probably not required to have more than 2 initially configured competitions. It's actually enough to have 2 for local testing purposes.

**Review by pelmegov — CHANGES_REQUESTED** [review]

**Comment by pelmegov** [comment]
@aalexeen any updates?

File: pom.xml (hunk 1 above)
Comment (aalexeen) [reply pom.xml]: deleted.

File: src/main/java/org/jacp/competitionservice/CompetitionserviceApplication.java (hunk 2 above)
Comment (aalexeen) [reply src/main/java/org/jacp/competitionservice/CompetitionserviceApplication.java]: done

File: src/main/resources/db/changelog/init/01-create-competition.yaml (hunk 4 above)
Comment (aalexeen) [reply src/main/resources/db/changelog/init/01-create-competition.yaml:8]: converted to yml

File: src/main/resources/application.properties (hunk 3 above)
Comment (aalexeen) [reply src/main/resources/application.properties]: converted to yml

File: src/main/resources/db/competition.csv (hunk 5 above)
Comment (aalexeen) [reply src/main/resources/db/competition.csv]: 3 left only.

File: src/main/resources/db/changelog/init/01-create-competition.yaml (hunk 4 above)
Comment (aalexeen) [reply src/main/resources/db/changelog/init/01-create-competition.yaml:8]: working on csv file format

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>3.1.2</version>
+        <relativePath/> <!-- lookup parent from repository -->
+    </parent>
+    <groupId>org.jacp</groupId>
+    <artifactId>competitionservice</artifactId>
```
Comment (saffchen) [inline pom.xml:12]: Minor: maybe use CamelCase on the artefactId and name?

File: src/main/resources/db/changelog/init/01-create-competition.yaml (hunk 4 above)
Comment (pelmegov) [reply src/main/resources/db/changelog/init/01-create-competition.yaml:8]: Could you please elaborate little bit?

**Comment by pelmegov** [comment]
@aalexeen please push new commits to the branch.

**Review by saffchen — APPROVED** [review]

**Review by pelmegov — APPROVED** [review]

**Pull request merged**